### PR TITLE
Allow doctrine/orm 3.x

### LINKED ISSUE #12: Allow doctrine/orm 3.x
Doctrine/orm 3 released on february 3 (https://github.com/doctrine/orm/releases/tag/3.0.0). Is it possible to update the constraint in composer.json to allow for this new version?

### PR DESCRIPTION
**Pull request by ToshY**
Fixes #12

### CODE REVIEW

**Comment by adrenalinkin** [comment]
@ToshY Hello! Do you have any idea why tests are failed

**Comment by ToshY** [comment]
> @ToshY Hello! Do you have any idea why tests are failed

I will take look. Not sure if it's related to the fix, but I can try to fix them.

**Comment by adrenalinkin** [comment]
@ToshY Try to run `composer require doctrine/annotations --dev` for fix Unit Tests

File: composer.json [hunk 1]
```diff
@@ -17,11 +17,12 @@
     ],
     "require": {
         "php": "^7.1 || ^8.0",
-        "doctrine/orm": "^2.3",
+        "doctrine/orm": "^2.3 || ^3.0",
         "symfony/deprecation-contracts": "^2.1 || ^3.0"
     },
     "require-dev": {
         "doctrine/doctrine-bundle": "^1.6 || ^2.0",
+        "friendsofphp/php-cs-fixer": "3.3.0",
```
Comment (adrenalinkin) [inline composer.json]: Don't add it as dependency
You can use this instruction to setup your local environment with predefined PHP-CS-Fixer
https://gist.github.com/adrenalinkin/35176a7b52b996666c4a36384fd536ad

**Comment by ToshY** [comment]
Couple of points I want to mention:

* Everything below PHP 8.1 is actually [EOL](https://www.php.net/supported-versions.php) so I changed constraint (still to `^8.0` as it introduced attributes).
  * This also explains the change to `attribute` instead of `annotation` for driver.
* I'm confused why `phpunit` is allowed as a dev dependecy and `php-cs-fixer` not? If it's not mentioned in the `composer.json` how would a contributor know that he/she even needs `php-cs-fixer` and which version to use?

I understand if this might be too big of a change to merge, and if so, I can always just publish the fork with updated constraints myself. 

Let me know.

**Comment by adrenalinkin** [comment]
@ToshY Hello! Thanks for your help with this contributing. I will accept it and then add some correction by myself to push new 3.0 version

> I'm confused why phpunit is allowed as a dev dependecy and php-cs-fixer not?

This is common practice 

**php-coveralls**
- Installed as a separate [composer.json in devtools](https://github.com/php-coveralls/php-coveralls/blob/master/dev-tools/composer.json) and [no dependencies in composer.json](https://github.com/php-coveralls/php-coveralls/blob/master/composer.json#L43) 

**Symphony**
- Installed as a side dependency. Here is the [Documentation](https://symfony.com/doc/current/contributing/code/standards.html#making-your-code-follow-the-coding-standards) and [no dependencies in composer.json](https://github.com/symfony/symfony/blob/7.1/composer.json#L124)

File: .github/workflows/unit-tests.yaml [hunk 2]
```diff
@@ -11,21 +11,17 @@ jobs:
...
-                    -   php: '7.2'
-                        symfony: '^4.0'
-                    -   php: '7.3'
-                        symfony: '^5.0'
-                    -   php: '7.4'
-                        symfony: '^5.0'
                     -   php: '8.0'
-                        symfony: '^6.0'
-                        coverage: 'pcov'
+                        symfony: '^5.4'
```
Comment (adrenalinkin) [inline .github/workflows/unit-tests.yaml:15]: Here should be flag 
`composer-flags: '--prefer-lowest'`

**Comment by ToshY** [comment]
Hey @adrenalinkin 

Thanks for clarifiying. I'm not aware of this practice to have a separate `composer.json` for just the devtools, that's kind of interesting. What's the benefit of doing it this way? Isn't it the same as putting your dev dependencies in the `require-dev` of your main `composer.json`? And if both `phpunit` and `php-cs-fixer` are devtools, why would `phpunit` be allowed in the `require-dev` section of the main `composer.json`, whereas `php-cs-fixer` should be separated to the other one?

I'm now mainly curious on this way of working. Do you know any blogs/articles/documentation/explanantion regarding the practice of having a separate `composer.json` for devtools (because I can't really seem to find it)?

**Comment by adrenalinkin** [comment]
@ToshY Hello! I'm faced this concept for the first time when i did contribution into php-coveralls. 
You can read more in this comment https://github.com/php-coveralls/php-coveralls/pull/325#discussion_r748887075
There also you can get more information about reason of this practice. Hope it's help.

My short explanation:
- `php-cs-fixer` - is just a tool for code style. The philosophy of this tool don't allow you to use set of versions. Project which will use `adrenalinkin/doctrine-naming-strategy` also can use `php-cs-fixer`. But that project **is not obliged** to be compatible with the same version of the `php-cs-fixer` as my library `adrenalinkin/doctrine-naming-strategy`.  
- `phpunit ` - the philosophy of this package allow to use set of versions and `adrenalinkin/doctrine-naming-strategy` **is obliged** to work with all set of phpunit version which defined in `require-dev`. In this case project will not have problems with dev dependencies in `require-dev` section.

**Comment by ToshY** [comment]
@adrenalinkin Thanks for taking the time to elaborate. I will also check out the link to further read into it.

---

Also appreciate the `3.0` release 👍

**Pull request merged**